docs/encyclopedia/application-message-passing.mdx

```diff
@@ -220,22 +220,40 @@ Every time the Workflow wakes up--generally, it wakes up when it needs to--it wi
 
 This execution is on a single thread–while this means you don’t have to worry about parallelism, you do need to worry about concurrency if you have written Signal and Update handlers that can block. These can run interleaved with the main Workflow and with one another, resulting in potential race conditions. These methods should be made reentrant.
 
+#### Initializing the Workflow first {#workflow-initializers}
+
+Initialize your Workflow's state before handling messages. This prevents reading uninitialized instance variables.
+
+To see why, refer to the [diagram](#message-handler-concurrency). It shows that your Workflow processes messages before the first run of your Workflow's main method.
+
+This happens in several scenarios, such as:
+
+- When using [Signal-with-Start](#signal-with-start), the message handler always runs first.
+- When your Worker experiences delays, such as during backlogs.
+- When messages arrive immediately after a Workflow continues as new but before it resumes.
+
+For all languages except Go and TypeScript, use your constructor to set up state. Annotate your constructor as an Workflow Initializer and take the same arguments as your Workflow's main method.
```

Comment:
Just so you know _why_ we use separate lines. 

- Improves version control diffs for easier line-by-line inspection.
- Focuses feedback and review discussions on single points, preventing overlap with other content that may need changes.
- Reduces overlap and merge conflicts with other tickets.
- Enhances the precision of our automated lint tooling, such as Vale (`vale /path/to/file.mdx`), allowing you to more easily locate and address issues by line number.

Reply:
TBH, while these points seem valid, I didn't really resonate with any of them because they don't improve the flow for the end reader.  I'm happy to make things harder for my reviews to make the prose flow better.
Keeping it as-is and would advocate for a users-first mindset here.